framework/db/src/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -969,7 +969,12 @@ public T findByUuidIncludingRemoved(final String uuid) {
     @Override
     @DB()
     public T findByIdIncludingRemoved(ID id) {
-        return findById(id, true, null);
+        if (_cache != null) {
+            final Element element = _cache.get(id);
+            return element == null ? findById(id, true, null) : (T)element.getObjectValue();
+        } else {
```

Comment:
@dahn if the ``_cache`` is not optional then I agree. It should probably be something like the following:

```
if (_cache == null) {
    throw new IllegalStateException("The cache has not been properly initialized for DAO " + this.getClass().getSimpleName());
}

final Element element = _cache.get(id);
return element == null ? findById(id, true, null) : (T) element.getObjectValue();
```

Would you agree?